Improve langchain autologging doc (#10930)
mlflow/mlflow: docs/source/llms/langchain/guide/index.rst

- **Seamless Metrics Recording**: Autologging effortlessly captures metrics for evaluating generated texts, as well as key objects such as llms and agents employed during inference.
- **Automated Input and Output Logging**: Autologging takes care of logging inputs and outputs of the LangChain model during inference. The recorded results are neatly organized into an inference_inputs_outputs.json file, providing a comprehensive overview of the model's inference history.

.. note::

>     To use MLflow LangChain autologging, please upgrade langchain to version 0.1.0 or higher.

Reviewer: ```suggestion
    To use MLflow LangChain autologging, please upgrade langchain to **version 0.1.0** or higher.
```

>     Currently you might need to manually install langchain_community>=0.0.16 to enable autologging artifacts and metrics. (we're working on improvements in the meantime)

Reviewer: ```suggestion
    Depending on your existing environment, you may need to manually install langchain_community>=0.0.16 in order to enable the automatic logging of artifacts and metrics. (this behavior will be modified in the future to be an optional import)
```

>     If autologging doesn't log artifacts as expected, please check the warning messages in stdout logs. 

Reviewer: ```suggestion
    If autologging doesn't log artifacts as expected, please check the warning messages in `stdout` logs. 
```

>     For langchain_community==0.0.16, please try to install textstat and spacy libraries manually, and restart your python environment.

Reviewer: ```suggestion
    For langchain_community==0.0.16, you will need to install the `textstat` and `spacy` libraries manually, as well as restarting any active interactive environment (i.e., a notebook environment). On Databricks, you can achieve this via executing `dbutils.library.restartPython()` to force the Python REPL to restart, allowing the newly installed libraries to be available.
```

> MLflow langchain autologging injects `MlflowCallbackHandler <https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/callbacks/mlflow_callback.py>`_ into the langchain model inference process, to log

Reviewer: ```suggestion
MLflow langchain autologging injects `MlflowCallbackHandler <https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/callbacks/mlflow_callback.py>`_ into the langchain model inference process to log
```
